[ADD] Payroll: Mexico localization (#14767)
odoo/documentation: content/applications/hr/payroll/payroll_localizations/mexico.rst

contributed by the employee, enter the additional amount in the :guilabel:`Extra Fixed Monthly
Contribution` field.

The last field corresponds to the :guilabel:`Type` selected. Enter the :guilabel:`Fixed Monetary
Fee`, :guilabel:`Percentage`, or  :guilabel:`Discount Factor` in the field.

Once all fields are configured, click the :guilabel:`Save & Close` button, and the credit is added
to the :abbr:`INFONAVIT (Instituto del Fondo Nacional de la Vivienda para los Trabajadores)`
section.

.. image:: mexico/infonavit.png
   :alt: The INFONAVIT section filled out for an employee.

.. important::
   Once :abbr:`INFONAVIT (Instituto del Fondo Nacional de la Vivienda para los Trabajadores)`
   credits are fully paid, the payroll officer **must** change the :guilabel:`Status` of each
   completed line item to :guilabel:`Closed`.

   If the payroll officer does **not** change the status, money will continue to be contributed to
   the employee's account in all subsequent paychecks.

Fonacot section
~~~~~~~~~~~~~~~

:abbr:`FONACOT (Fondo Nacional para el Consumo de los Trabajadores)` is a Mexican government program

> that provides employees with a lower-cost loan to cover a variety of situations, from home
> improvements, to education, to emergencies.

Reviewer: slight, optional brevity suggestion
```suggestion
that provides employees with with low-cost loans for various needs, such as home improvements,
education, or emergencies.
```

The :abbr:`FONACOT (Fondo Nacional para el Consumo de los Trabajadores)` section is where employee
repayments are configured.

Add an Infonavit credit
***********************

To add a line to the :abbr:`FONACOT (Fondo Nacional para el Consumo de los Trabajadores)` section,
click :guilabel:`Add a line`, and a blank line loads. The :guilabel:`Status` is set to :guilabel:`In
Progress`, by default.

Enter the monthly payment amount in the :guilabel:`Import` field. If the employee wishes to make any
additional payments each month, enter the value in the :guilabel:`Extra Monthly Contribution` field.

.. important::
   Once :abbr:`FONACOT (Fondo Nacional para el Consumo de los Trabajadores)` loans are paid off, the
   payroll officer **must** change the :guilabel:`Status` of each completed line item to
   :guilabel:`Closed`.

   If the payroll officer does **not** change the status, money will continue to be taken out of the
   employee's account in all subsequent paychecks.

Payroll configuration
=====================

Several sections within the **Payroll** app installs a Salary Structure, Structure Type, Rules, and

> Parameters specific to Mexico.

Reviewer: Are Salary structure, type, rules, and parameters fields? If so, let's italicize these. If we're referring to them generally as a concept, let's use lowercase

Prior to running payroll, the :abbr:`CFDI (Comprobante Fiscal Digital por Internet)` must also be
configured.

CFDI configuration
------------------

The :abbr:`CFDI (Comprobante Fiscal Digital por Internet)` is a **required** electronic tax receipt
that acts as a digital invoice. The :abbr:`CFDI (Comprobante Fiscal Digital por Internet)` **must**
be certified by a :abbr:`PAC (Authorized Certification Provider)` *before* being submitted to the
Mexican Tax Administration (:abbr:`SAT (Servicio de Administración Tributaria)`).

This document serves as proof of income, expenses, and payroll, and requires both a :abbr:`UUID
(Universally Unique Identifier)` and a digital signature, to comply with Mexican tax laws.

First, navigate to :menuselection:`Payroll app --> Configuration --> Settings`, and scroll to the
:guilabel:`CFDI` section.

> Enter the :guilabel:`Employer` :abbr:`IMSS (Instituto Mexicano del Seguro Social)` :guilabel:`ID` in

Reviewer: > :guilabel:`Employer` :abbr:`IMSS (Instituto Mexicano del Seguro Social)` :guilabel:`ID`

Is "Employer IMSS ID" the field? While I think we should be consistent to do guilabel on the whole field to match the UI, it's useful to define the abbreviation. I guess the question is: do Mexican payroll accountants know what IMSS stands for? If it's not a commonly used abbreviation, I see the value in defining the abbreviation. I recommend following up with Miguel on this! I think he'd best know

Author: True- I had a hard time with this, because I wanted both - I was hoping the gui would bold the abbreviation, but it doesn't! But yes, it is the whole field name, so I'll change that.

the corresponding field, and select the appropriate :guilabel:`Risk Type` using the drop-down menu.
If no risk is associated with the company, select :guilabel:`Does Not Apply`.

.. note::

>    Mexico assigns different :guilabel:`Risk Types` based on a variety of factors, such as
>    cross-referencing prior submitted documents from both employees and employers. Refer to the
>    payroll officer to determine the correct risk type.

Reviewer: optional!
```suggestion
   Mexico assigns different :guilabel:`Risk Types` based on several factors, including information
   cross-referenced from previously submitted employee and employer documents. Consult your payroll
   officer to determine the correct risk type.
```

note: I'm having a hard time deciding whether we use "your" here. I know we typically want to avoid it, but I think it's more clear and concise this way, especially since we're telling the reader to rely on a person's expertise

- :guilabel:`601.27.01 Contributions to infonavit`
- :guilabel:`601.28.01 Contributions to the SAR`
- :guilabel:`201.01.02 Employee Reimbursement`
- :guilabel:`205.06.02 Other Various Short-Term Credits (Fonacot)`
- :guilabel:`210.03.01 Provision for bonus to pay`
- :guilabel:`210.04.01 Provision for savings fund to pay`
- :guilabel:`211.03.01 Provision for infonavit to pay`
- :guilabel:`601.12.01 Bonus`
- :guilabel:`601.16.02 Transport (gasoline vouchers)`
- :guilabel:`601.16.01 Transport (support)`
- :guilabel:`601.74.01 Commissions on sales`

.. note::
   The :abbr:`CoA (Chart of Accounts)` configuration is done by default when a company is located in
   Mexico. The account codes and names can be edited to suit the company's needs. If there is no
   :abbr:`CoA (Chart of Accounts)` account associated with a salary rule (used in a salary
   structure), Odoo uses the account `Salary Expenses` to create the journal entry, regardless of
   the nature of the move.

If everything seems correct on the journal entry draft, post the journal entries.

.. image:: mexico/post-entries.png
   :alt: Journal entries posted.

.. _payroll/register-mx:

Register Payments
-----------------

> After the :ref:`journla entries <payroll/journal-mx>` are validated, Odoo can generate payments.

Reviewer: ```suggestion
After the :ref:`journal entries <payroll/journal-mx>` are validated, Odoo can generate payments.
```

.. important::
   To generate payments from payslips,employee's **must** have a *trusted* bank account. If the
   employee's bank account is *not* marked as `trusted`, CFDI files **cannot** be generated through
   Odoo.

.. image:: mexico/paid.png
   :alt: Payslips with a status of paid.

Generate CFDI XML files
-----------------------

After payslips have bee processed and paid, a CFDI XML file must be generated, which is then
electronically submitted to Mexico's :abbr:`SAT (Servicio de Administración Tributaria)`.

> To create the CFDI file, open the individual payslip, and click the :guilabel:`Generate CFDI`

Reviewer: suuuper small but i think we can add a abbreviation to CFDI (: 

XML can be left alone haha

       - l10n_mx_edi
     - Adds the required rules and parameters to manage local payroll calculations.

.. tip::
   Installing the :guilabel:`Mexico - Payroll CFDI` module automatically installs both the
   :guilabel:`Mexico - Payroll` and :guilabel:`Mexico - Payroll with Accounting` modules.

.. seealso::
   :doc:`Configure the Mexico fiscal localization <../../../finance/fiscal_localizations/mexico>`

General configurations
======================

First, the company must be configured. Navigate to :menuselection:`Settings app --> Users &
Companies --> Companies`. From the list, select the desired company, and configure the following
fields:

- :guilabel:`Company Name`: Enter the business name in this field.
- :guilabel:`Address`: Complete the full address, including the :guilabel:`City`, :guilabel:`State`,
  :guilabel:`Zip Code`, and :guilabel:`Country`.

  .. important::
     The state selected in the company's address is associated as the *work address* by default for
     the employee, and the one used to calculate payroll taxes.

- :guilabel:`RFC`: Enter the company's :abbr:`RFC (Registro Federal de Contribuyentes, or Federal
  Taxpayer Registry)` number.

> - :guilabel:`Company ID` : Enter the business's registry number.

Author: I mentioned it is not used and it should be blank- only because I like ot epxlain ALL field,s so if it shouldn't be filled i, I added that info.

     overridden on individual employee records if exceptions are needed.

- :guilabel:`Wage`: Enter the amount and frequency the employee is paid.
- :guilabel:`Contract Type`: Determines how the employee is paid and classified, such as
  :guilabel:`Indefinite Term Employment Contract`, :guilabel:`Hourly Wage Employment`,
  :guilabel:`Seasonal Employment Contract`.

  .. important::
     The :guilabel:`Contract Type` field is visible on payslips.

- :guilabel:`Pay Category`: Select :guilabel:`Mexico: Employee.` for this field. This defines when
  the employee is paid, their default working schedule, and the work entry type it applies to.
- :guilabel:`Regime Type`: Select the correct regime type (Régimen de Contratación) for the
  employee. The selected regime type determines the employee's benefits, taxes, and social security
  contributions.

Schedule section
~~~~~~~~~~~~~~~~

- :guilabel:`Work Entry Source`: Defines how :doc:`work entries <../work_entries>` are generated for
  payroll during the specified pay period. The options are:

  - :guilabel:`Working Schedule`: Based on the employee's assigned :ref:`working schedule
    <payroll/working-times>` (e.g., 40 hours per week).

>   - :guilabel:`Attendances`: Based on :doc:`approved checked-in hours

Author: Is it compatible with the Planning app as well? Or ONLY working schedules?

- :guilabel:`Meal Vouchers`: Enter the monthly amount paid to the employee towards food expenses.
- :guilabel:`Transport Vouchers`: Enter the monthly amount paid to the employee towards
  transportation, such as bus fare or parking fees.
- :guilabel:`Gasoline Vouchers`: Enter the monthly amount paid to the employee for their vehicle's
  gas.

.. note::
   The monetary amount for the vouchers is the *monthly amount* given to employees. If the
   :guilabel:`Payment Period` is set to :guilabel:`In the period`, Odoo calculates the voucher
   amount for each pay period.

   For example, if an employee is paid weekly, and their :guilabel:`Meal Voucher` is set to $100,
   the employee receives $25 in every weekly paycheck.

.. image:: mexico/payroll-tab.png
   :alt: The payroll tab of an employee form, filled out for Mexico.

Infonavit / Fonacot tab
-----------------------

Infonavit section
~~~~~~~~~~~~~~~~~

:abbr:`INFONAVIT (Instituto del Fondo Nacional de la Vivienda para los Trabajadores)` is a
government institution that assists employees with affordable mortgage credits, which can be used to
buy, build, or remodel a home, or pay off an existing mortgage.

> Employers are required to contribute 5% of each employees' salary into their individual employee's

Author: OK thank you! Can I leave the "...government institution that assists employees with affordable mortgage credits, which can be used to buy, build, or remodel a home, or pay off an existing mortgage." sentence, or that, too?

housing subaccount. :abbr:`INFONAVIT (Instituto del Fondo Nacional de la Vivienda para los
Trabajadores)` credits make lower interest rates with flexible terms available to employees.

Add an Infonavit credit
***********************

To create an :abbr:`INFONAVIT (Instituto del Fondo Nacional de la Vivienda para los Trabajadores)`
credit, click :guilabel:`Add a line`, and a :guilabel:`Create an Infonavit Credit` pop-up window
loads.

By default, the :guilabel:`Status` field is populated with :guilabel:`In Progress`. This indicates
the credit is active.

Next, select the :guilabel:`Type` of credit, either a :guilabel:`Fixed Monetary Fee`, a
:guilabel:`Percentage`, or a :guilabel:`Discount Factor`. After this field is configured, a
corresponding field appears on the form.

> Enter the :guilabel:`Monthly Insurance` amount the *company* contributes to the employee's housing

Author: Is it still paid by the company? I am not sure how to repohrase this, as I don't completely understand what is happening/who is paying what, based on what, for what. Could you help me rephrase this so it is accurate?

repayments are configured.

Add an Infonavit credit
***********************

To add a line to the :abbr:`FONACOT (Fondo Nacional para el Consumo de los Trabajadores)` section,
click :guilabel:`Add a line`, and a blank line loads. The :guilabel:`Status` is set to :guilabel:`In
Progress`, by default.

Enter the monthly payment amount in the :guilabel:`Import` field. If the employee wishes to make any
additional payments each month, enter the value in the :guilabel:`Extra Monthly Contribution` field.

.. important::
   Once :abbr:`FONACOT (Fondo Nacional para el Consumo de los Trabajadores)` loans are paid off, the
   payroll officer **must** change the :guilabel:`Status` of each completed line item to
   :guilabel:`Closed`.

   If the payroll officer does **not** change the status, money will continue to be taken out of the
   employee's account in all subsequent paychecks.

Payroll configuration
=====================

Several sections within the **Payroll** app installs a Salary Structure, Structure Type, Rules, and
Parameters specific to Mexico.

Prior to running payroll, the :abbr:`CFDI (Comprobante Fiscal Digital por Internet)` must also be
configured.

> CDFI configuration

Author: That's when my eyes started to cross, i think! =D

.. note::
   To cut down on the payroll officer's time, it is typical to process payslips in batches, either
   by wage type (fixed salary vs hourly), pay schedule (weekly, bi-weekly, monthly, etc.),
   department (direct cost vs. administration), or any other grouping that best suits the company.

The process of running payroll includes different actions that need to be executed to ensure that
the amount withheld for funds and government institutions is correct, the amount that the employee
receives as their net salary is correct, and the computation of hours worked reflects the employee's
actual hours worked, among others.

When running a payroll batch, check that the period, company, and employees included are correct
*before* starting to analyze or validate the data.

Once the payslips are drafted, review them for accuracy. Check the :guilabel:`Worked Days` and
:guilabel:`Salary Inputs` tabs, and ensure the listed worked time is correct, as well as any other
inputs. Add any missing inputs, such as commissions, bonuses, and :abbr:`ISR (Impuesto Sobre la
Renta)` adjustments that are missing.

> Next, check the various totals (gross pay, vouchers, IMSS contributions, :abbr:`CEAV (Comisión

Author: OMG! I was googling everything and trying to look up what all the institutions and acronyms are- so I could better understand this, AND write it correctly. No wonder I picked the wrong one!
